feat: Updates of `avm/res/network/express-route-circuits`

thecmdradama opened the pull request:
## Description

- Adds support for an optional list of authorizations.
- Adds output for authorizations as a flattened map.
- Adds output for the service provider provisioning state property.

<!--
>Thank you for your contribution !
> Please include a summary of the change and which issue is fixed.
> Please also include the context.
> List any dependencies that are required for this change.

Fixes #123
Fixes #456
Closes #123
Closes #456
-->

## Pipeline Reference

<!-- Insert your Pipeline Status Badge below -->

| Pipeline |
| -------- |
|  [![avm.res.network.express-route-circuit](https://github.com/thecmdradama/bicep-registry-modules/actions/workflows/avm.res.network.express-route-circuit.yml/badge.svg)](https://github.com/thecmdradama/bicep-registry-modules/actions/workflows/avm.res.network.express-route-circuit.yml)  |

## Type of Change

<!-- Use the checkboxes [x] on the options that are relevant. -->

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [x] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [x] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

<!--  Please keep up to date with the contribution guide at https://aka.ms/avm/contribute/bicep -->

thecmdradama commented:
Finally fixed a few bugs and added a missing output for authorizations that should be there. Ready for review.

AlexanderSehr commented on avm/res/network/express-route-circuit/main.bicep:
```diff
@@ -245,6 +253,16 @@ output name string = expressRouteCircuit.name
+
+@description('The authorizations for the express route circuit.')
+output authorizations array = map(expressRouteCircuit.properties.authorizations, (authorization, id) => {
+  id: authorization.id
```
I guess this is a resourceId? If the case, I'd recommend renaming it to `resourceId`, just to be as explicit as possible

thecmdradama replied:
Yes, that would be the `resourceId` for the Authorization so that does indeed make sense to rename it to `resourceId` from just `id`. Fixed

AlexanderSehr replied:
Much appreciated 💪  Thank you @thecmdradama

AlexanderSehr commented on avm/res/network/express-route-circuit/main.bicep:
```diff
@@ -245,6 +253,16 @@ output name string = expressRouteCircuit.name
+output authorizations array = map(expressRouteCircuit.properties.authorizations, (authorization, id) => {
+  id: authorization.id
+  name: authorization.name
+  key: authorization.properties.authorizationKey
```
Would the authorization key not be a sensitive bit of information? If so, mind you, that Bicep (as of today) does not support `secure()` outputs

thecmdradama replied:
I don't believe that it is considered to be sensitive as they are shown in the portal as plaintext.

AlexanderSehr replied:
Well... storage access keys are too 😄 The portal is unfortuantely not a reliable source.

@arnoldna, what's your take on this? If we end up agreeing it **is** in fact sensitive, we only have 2 choices:
1. Remove it from the output and implement AVM's KeyVault export feature to store this item in a key vault during dpeloyment - OR
2. Remove it from the output and ask users to fetch the key using e.g. an `existing` resource reference to fetch the key on the fly (or maybe there even is a function, like `listKeys()`)

thecmdradama replied:
Sorry what I meant was there's no visual obfuscation with the authorisation keys (I.e. a show/hide button). Like with storage access keys, but I get your point about not replying on the portal.

Yeah, if it is considered to be sensitive. I think the second option is better and I can picture how doing an existing resource reference for `microsoft.network/expressroutecircuits/authorisations` would work.

arnoldna replied:
I am inclined to believe that it is sensitive information, and we shouldn't be outputting it. I am in favor of the second option as well.

thecmdradama replied:
@arnoldna thanks for the confirmation. Will remove authorisations from outputs.

thecmdradama commented:
Hey @AlexanderSehr I've added to the description for the `authorizationNames` parameter around fetching the `authorizationKey` after creation, Is that wording ok? If so, is this good to go?

AlexanderSehr commented on avm/res/network/express-route-circuit/main.bicep:
```diff
@@ -60,6 +60,9 @@ param location string = resourceGroup().location
 param allowClassicOperations bool = false
 
+@description('Optional. List of names for ExpressRoute circuit authorizations to create. To fetch the `authorizationKey` for the authorization, use the `existing` resource reference for `Microsoft.Network/expressRouteCircuits/authorizations`.')
+param authorizationNames array = []
```
```suggestion
param authorizationNames string[] = []
```

AlexanderSehr commented:
> Hey @AlexanderSehr I've added to the description for the `authorizationNames` parameter around fetching the `authorizationKey` after creation, Is that wording ok? If so, is this good to go?

Hey @thecmdradama,
not much more too add from my side. I added one more suggestion (for which I'd ask you to re-run the `Set-AVMModule` utility after applying it), but the ultimate approval would lay with the module owner @arnoldna :)

thecmdradama commented:
> > Hey @AlexanderSehr I've added to the description for the `authorizationNames` parameter around fetching the `authorizationKey` after creation, Is that wording ok? If so, is this good to go?
> 
> Hey @thecmdradama, not much more too add from my side. I added one more suggestion (for which I'd ask you to re-run the `Set-AVMModule` utility after applying it), but the ultimate approval would lay with the module owner @arnoldna :)

Ah, thanks for that. Indeed, that makes more sense to enforce an Array of Strings than just an Array.

Re-ran Set-AVMModule and a new workflow run as well

[![avm.res.network.express-route-circuit](https://github.com/thecmdradama/bicep-registry-modules/actions/workflows/avm.res.network.express-route-circuit.yml/badge.svg?branch=feature%2Ferc-updates)](https://github.com/thecmdradama/bicep-registry-modules/actions/workflows/avm.res.network.express-route-circuit.yml)
